pkg/gateway/kubernetes.go

```diff
@@ -11,21 +11,23 @@ import (
 	"strings"
 )
 
-type ServiceMapperConfig struct {
+type kubernetesReverseProxyConfigGetterConfig struct {
```

Comment:
Yes, it totally could. I think some refactoring will fall out here as we move to writing data into the annotations.